src/app/clusters/ota-requestor/DefaultOTARequestorDriver.cpp

```diff
@@ -45,6 +45,7 @@ namespace {
 using namespace app::Clusters::OtaSoftwareUpdateRequestor;
 using namespace app::Clusters::OtaSoftwareUpdateRequestor::Structs;
 
+constexpr uint8_t kMaxInvalidSessionRetries        = 3;  // Max # of query image retries to perform on invalid session error
```

Comment:
Just retry once. Currently, there's two different scenarios where we can have an invalid session:
1. When provider was restarted since the last successful QueryImage - a second attempt *should* succeed
2. When provider is truly not reachable - a second attempt (or any further attempts) would fail

We're banking on that invalid session means CASE got torn down. If on a second attempt, we still get a timeout error (resulting in IdleStateReason::kInvalidSession), then that should be case (2) above and we shouldn't retry again
```suggestion
constexpr uint8_t kMaxInvalidSessionRetries        = 1;  // Max # of query image retries to perform on invalid session error
```